sky/data/storage.py

```diff
@@ -1871,6 +1904,540 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+                ('azure', 'resource_group'),
+                f'sky{common_utils.get_user_hash()}')
+            # check if the resource group with the given name already exists.
+            try:
+                self.resource_client.resource_groups.get(
+                    self.resource_group_name)
+            except azure.exceptions().ResourceNotFoundError as e:
+                if 'Code: ResourceGroupNotFound' in e.message:
+                    self.resource_client.resource_groups.create_or_update(
+                        self.resource_group_name, {'location': self.region})
+            # check if the storage account name already exists under the given
+            # resource group name.
+            try:
+                self.storage_client.storage_accounts.get_properties(
+                    self.resource_group_name, self.storage_account_name)
+            except azure.exceptions().ResourceNotFoundError as e:
+                if 'Code: ResourceNotFound' in e.message:
+                    try:
+                        creation_response = self.storage_client.storage_accounts.begin_create(
+                            self.resource_group_name, self.storage_account_name, {
+                                'sku': {
+                                    'name': 'Standard_GRS'
+                                },
+                                'kind': 'StorageV2',
+                                'location': self.region,
+                                'encryption': {
+                                    'services': {
+                                        'blob': {
+                                            'key_type': 'Account',
+                                            'enabled': True
+                                        }
+                                    },
+                                    'key_source': 'Microsoft.Storage'
+                                },
+                            }).result()
+                        # Assigning Storage Blob Data Owner role of the storage account
+                        # Reference: https://github.com/Azure/azure-sdk-for-python/issues/35573 # pylint: disable=line-too-long
+                        authorization_client = data_utils.create_az_client(
+                            'authorization')
+                        graph_client = data_utils.create_az_client('graph')
+                        async def get_object_id():
+                            import logging
+                            original_level = logging.getLogger("httpx").getEffectiveLevel()
+                            # silencing the INFO level response log from httpx request
+                            logging.getLogger("httpx").setLevel(logging.WARNING)
+                            user = await graph_client.users.with_url(
+                                'https://graph.microsoft.com/v1.0/me').get()
+                            logging.getLogger("httpx").setLevel(original_level)
```

Comment:
Could we do `httx_logger=logging.getLogger("httpx")` and use this variable in the following?

Reply:
Done.